src/sentry/static/sentry/app/components/group/inboxBadges/inboxReason.tsx

```diff
@@ -30,6 +31,46 @@ function InboxReason({inbox, fontSize = 'sm'}: Props) {
       ? `More than ${Math.round(count / EVENT_ROUND_LIMIT)}k`
       : `${count}`;
 
+  const getTooltipDescription = ({
```

Comment:
we could move this function out of InboxReason or we could also write it like below:


```
function InboxReason({inbox, selection, fontSize = 'sm'}: Props) {
  const {reason, reason_details, date_added: dateAdded} = inbox;

  const getCountText = (count: number) =>
    count > EVENT_ROUND_LIMIT
      ? `More than ${Math.round(count / EVENT_ROUND_LIMIT)}k`
      : `${count}`;

  function getTooltipDescription() {
    const {until, count, window, user_count, user_window} = reason_details;
    const {datetime} = selection;
    const {utc} = datetime;

    if (until) {
      // Was ignored until `until` has passed.
      //`until` format: "2021-01-20T03:59:03+00:00"
      return t('Was ignored until %(window)s', {
        until: getFormattedDate(until, 'MMM D, YYYY LT', {local: !utc}),
      });
    }

    if (count) {
      // Was ignored until `count` events occured
      // If `window` is defined, than `count` events occurred in `window` minutes.
      // else `count` events occured since it was ignored.
      if (window) {
        return tct('Was ignored until it occured ([count]) time(s) in [duration]', {
          count: getCountText(count || 0),
          duration: getDuration((window || 0) * 60, 0, true),
        });
      }

      return tct('Was ignored until it occured ([count]) time(s)', {
        count: getCountText(count || 0),
      });
    }

    if (user_count) {
      // Was ignored until `user_count` users were affected
      // If `user_window` is defined, than `user_count` users affected in `user_window` minutes.
      // else `user_count` events occured since it was ignored.
      return t('Was ignored until it affected %(user_count)s user(s)%(user_window)s', {
        user_count: getCountText(user_count || 0),
        user_window: user_window
          ? ' in ' + getDuration((user_window || 0) * 60, 0, true)
          : '',
      });
    }

    return undefined;
  }
```

Reply:
Switched it to a function. Not sure what the benefit of that is though, could you tell me?

Reply:
we use Babel to transpile the ES6 features, so if you write for example:

`const test = () => {} `

it will be transpiled to:

`var test = function test() {}`

while it could be:

`function test() {}`

because of that, I am preferring to write functions directly. I only use const when I need to use `this` or when I want something to be immutable.